src/unix/fs.c

```diff
@@ -217,6 +217,54 @@ static ssize_t uv__fs_mkdtemp(uv_fs_t* req) {
 }
 
 
+static int uv__fs_mkstemp(uv_fs_t* req) {
+  int r;
+  int (*mkostemp_function)(char*, int);
+  char *path = (char*) req->path;
+  int path_length = strlen(path);
+
+  /* EINVAL can be returned for 2 reasons:
+       1. The template's last 6 characters were not XXXXXX
+       2. open() didn't support O_CLOEXEC
+     We want to avoid going to the fallback path in case
+     of 1., so it's manually checked before. */
+  if (path_length < 6 ||
+      strcmp(path + (path_length - 6), "XXXXXX"))
+    return -EINVAL;
+
+  *(int**)(&mkostemp_function) = dlsym(RTLD_DEFAULT, "mkostemp");
+
+  /* We don't care about errors, but we do want to clean them up.
+     If there has been no error, then dlerror() will just return
+     NULL. */
+  dlerror();
+
+  if (mkostemp_function != NULL) {
+    return mkostemp_function(path, UV__O_CLOEXEC);
```

Comment:
this could still fail with EINVAL if `open` doesn't support `O_CLOEXEC`, so you need to fallback to `mkstemp`